### ISSUE TITLE
Unable to mark event as favourite through toolbar in event details 

### ISSUE THREAD

**Issue by ShridharGoel** [fossasia/open-event-android#1692]
**Describe the bug**
Unable to mark event as favourite through toolbar in event details 

**GIF**
![ezgif com-video-to-gif (67)](https://user-images.githubusercontent.com/35566748/56923374-ff9d4f80-6ae7-11e9-8c96-53b05ed1404a.gif)

**Would you like to work on the issue?**
Open to all.

**Comment by liveHarshit** [comment]
@ShridharGoel I'm not able to reproduce this issue. The issue is that event marked as favorite but recycle view is not updating after going back.

**Comment by ShridharGoel** [comment]
For me, the like icon is changing back to unliked state as in the GIF.

**Comment by ShridharGoel** [comment]
Will try again on latest commits

**Comment by ShridharGoel** [comment]
The issue is still there, please check @liveHarshit @anhanh11001

**Comment by anhanh11001** [comment]
Yes, I can still reproduce it, i'll take a look at it

**Comment by anhanh11001** [comment]
@ShridharGoel  Can you check if this issue is still reproducible. I think it is fixed now by #1707. I guess the reason was when calling this method to load the flowable from the database, we missed `.distinctUntilChanged()` so the UI constantly keeps updating (like after every 1 second) by following the LiveData, which led to weird UI behaviour in EventDetailsFragment

Now:
```
fun loadEvent(id: Long) {
        if (id == -1L) {
            mutablePopMessage.value = resource.getString(R.string.error_fetching_event_message)
            return
        }
        compositeDisposable += eventService.getEvent(id)
            .withDefaultSchedulers()
            .distinctUntilChanged() // This was missed before
            .doOnSubscribe {
                mutableProgress.value = true
            }.doFinally {
                mutableProgress.value = false
            }.subscribe({
                mutableEvent.value = it
            }, {
                Timber.e(it, "Error fetching event %d", id)
                mutablePopMessage.value = resource.getString(R.string.error_fetching_event_message)
            })
    }
```

**Comment by ShridharGoel** [comment]
Fixed.
